elastic/apm-server
# Make ecs.version a constant_keyword field

## Motivation/summary

Instead of adding `ecs.version` as a field in every document written by APM Server, make it a constant_keyword field and set its version in the data stream mappings. This means that `ecs.version` will not be available in `_source`, and will only be available through the Elasticsearch fields API. Nothing uses `ecs.version` yet, so it doesn't matter in practice.

The data stream's ingest pipeline is responsible for upgrading documents to the expected format, so it follows that the integration package (ingest pipeline, mappings) should define the ECS version. Given that nothing currently relies on `ecs.version`, the pipelines now simply drop any `ecs.version` field defined in the incoming document, in favour of the `constant_keyword` value.

We're currently pinning to a recent commit of ECS due to our reliance on the unreleased `device.*` fields. When they are released, we should update to the released version.

## Checklist

~- [ ] Documentation has been updated~

## How to test these changes

1. Install the integration, send some data to APM Server
2. Ensure the documents have `ecs.version` set

## Related issues

Closes #9043
